WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/details/BackupDownloadDetailsFragment.kt

```diff
@@ -30,24 +39,39 @@ class BackupDownloadDetailsFragment : Fragment() {
         val nonNullActivity = requireActivity()
         (nonNullActivity.application as WordPress).component().inject(this)
 
+        initRecyclerView()
+        initViewModel()
+    }
+
+    private fun initRecyclerView() {
+        recycler_view.layoutManager = LinearLayoutManager(context, RecyclerView.VERTICAL, false)
+        initAdapter()
+    }
+
+    private fun initAdapter() {
+        recycler_view.adapter = BackupDownloadDetailsAdapter(uiHelpers)
+    }
+
+    private fun initViewModel() {
         parentViewModel = ViewModelProviders.of(requireActivity(), viewModelFactory)
                 .get(BackupDownloadViewModel::class.java)
 
         viewModel = ViewModelProviders.of(this, viewModelFactory)
                 .get(BackupDownloadDetailsViewModel::class.java)
 
-        setupViews()
-        setupObservers()
-        // todo: annmarie - if something needs to be passed to VM, do it on start
-        viewModel.start()
-    }
+        viewModel.uiState.observe(viewLifecycleOwner, Observer { uiState ->
+            when (uiState) {
+                is Loading -> ToastUtils.showToast(requireContext(), "Implement loading")
+                is Content -> showContent(uiState)
+                is Error -> ToastUtils.showToast(requireContext(), "Implement Error")
+            }
+        })
 
-    private fun setupViews() {
-        // TODO: annmarie implement setupViews
+        viewModel.start()
```

Comment:
Warning (⚠️ ): Although the rotation should work, this line gets triggered twice on a configuration change. First ignores the `start()` (as expected) and the next one doesn't (new instance). As such, this defaults every checkbox to their default state from the provider.

This might have to do with the fact that the parent activity shows the fragment during the `setupViewModel()`. Can you check that, I know this is a temporary start, but testing the solution on configuration change seems to not work, which might make it harder to test any additional solution going forward. Wdyt?

Reply:
@ParaskP7 - I call out in the notes that rotation isn't addressed in this PR. Thanks for the great detail though, I'll make sure to review when I start to dig into the details. 🙇